hs/src/Reach/Connector.hs

```diff
@@ -38,6 +39,9 @@ data ConnectorError
   = Err_IntLiteralRange Integer Integer Integer
   deriving (Eq, Generic, ErrorMessageForJson, ErrorSuggestions)
 
+instance HasErrorCode ConnectorError where
+  errCode e = "RC" <> leftPad 4 '0' (show $ gconIndex e)
```

Comment:
You should make a helper in `Reach.Util` for this pattern AND you need to make a comment on the constructors that the indexes are now part of the external interface and cannot be changed. That's dangerous though. There *is* going to be a day when we put them in the wrong order, delete one, or something like that. I think it might be better to have them written out by hand. `HasErrorCode` will provide a default implementation of `errCode` that looks at the `errPrefix` and the `errIndex` and we manually type out a bunch of numbers to get `errIndex`